app/controllers/hyrax/handles_controller.rb

```diff
@@ -0,0 +1,80 @@
+module Hyrax
+  class HandlesController < ApplicationController
```

Comment:
I think you need a before action instead of the "if checks" on your methods.. you don't want the methods to proceed if handle_id is invalid and this isolates your validation checking to a single simple place.
```
before_action :validate_handle_id, only: [:handle_show, :handle_download]
...

private 
def validate_handle_id
  if params[:handle_id] != "1957"
    puts ..
    redirect_to '/404/file_not_found'
  end
end
```

Reply:
..and use your logging service, so we can get notification about this sort of traffic?